code/modules/fishing/sources/_fish_source.dm

```diff
@@ -53,6 +53,10 @@ GLOBAL_LIST_INIT(specific_fish_icons, zebra_typecacheof(list(
 	var/catalog_description
 	/// Background image name from /datum/asset/simple/fishing_minigame
 	var/background = "background_default"
+	/// It true, repeated and large explosions won't be as efficient. This is usually meant for global fish sources.
+	var/explosive_malus = FALSE
+	/// If explosive_malus is true, this will be used to keep track of the turfs where an explosion happened for when we'll spawn the loot.
+	var/list/exploded_turfs
```

Comment:
I was testing this pr, putting gibtonite at the edge of a lavaland lava river and having it explode for fish- I could only do this about once, before I could no longer get fish from the lava with any explosive I tried.
My first thought is, I don't think this ever gets reset anywhere? But I haven't tried to debug much yet.

Reply:
OH, YOU'RE RIGHT! I TOTALLY FORGOT!